javascript.js

```diff
@@ -0,0 +1,146 @@
+const validateForm = () => {
+  const tasks = JSON.parse(localStorage.getItem('datas')) ?? [];
+  const task = document.getElementById('task-input');
+  if (task && task.value !== '') {
+    const newEntry = {
+      index: tasks.length + 1,
+      completed: false,
+      description: task && task.value,
+    };
+    task.value = '';
+
+    tasks.push(newEntry);
+    localStorage.setItem('datas', JSON.stringify(tasks));
+
+    return false;
+  }
+
+  return true;
+};
+
+
+const deleteTask = (i) => {
+  let tasks = JSON.parse(localStorage.getItem('datas'));
+  tasks = tasks.filter((task) => task.description !== tasks[i].description);
+  for (let i = 0; i < tasks.length; i += 1) {
```

Comment:
So far so good, But I have something to add on this. Variable names should be meaningful and descriptive. places, I kindly request you to improve variable names for better readability. For example, i in the deleteTask and showTask functions could be renamed to taskIndex or something similar to make it clearer.